lib/beaker/dsl/helpers/puppet_helpers.rb

```diff
@@ -327,8 +327,14 @@ def bounce_service host, service, curl_retries = nil, port = nil
             apachectl_path = host.is_pe? ? "#{host['puppetsbindir']}/#{service}" : service
             host.exec(Command.new("#{apachectl_path} graceful"))
           else
-            host.exec puppet_resource('service', service, 'ensure=stopped')
-            host.exec puppet_resource('service', service, 'ensure=running')
+            result = host.exec(Command.new("service #{service} reload"),
+                               :acceptable_exit_codes => [0,1,3])
+            if result.exit_code == 0
+              return result
+            else
+              host.exec puppet_resource('service', service, 'ensure=stopped')
```

Comment:
This approach allows the helper to fall back to using the existing stopped/running logic in the event that the `reload` command fails - which could happen if the service had not been started when the `bounce_service` call is made or if the service does not support the `reload` command.  For the latter, this would allow a newer Beaker which has this logic to still be able to restart the service properly even when run against older versions of Puppet Server, when necessary.